### PR TITLE
Added solarized theme

### PR DESCRIPTION
**Pull request by shadmansaleh**
ported from lightline

### CODE REVIEW

File: lua/lualine/themes/solarized.lua [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  base3   =  '#fdf6e3',
+  yellow  =  '#b58900',
+  orange  =  '#cb4b16',
+  red     =  '#dc322f',
+  magenta =  '#d33682',
+  violet  =  '#6c71c4',
+  blue    =  '#268bd2',
+  cyan    =  '#2aa198',
+  green   =  '#859900',
+
```
Comment (hoob3rt) [inline lua/lualine/themes/solarized.lua:27]: I'm only gonna mention it here but it applies to all new theme pr's. Remove these awful empty lines please :)

File: lua/lualine/themes/solarized.lua [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+
+}
+if vim.o.background == 'light' then
```
Comment (hoob3rt) [inline lua/lualine/themes/solarized.lua:31]: While this is a really neat trick I think there should be two themes: `solarized_dark` and `solarized_light`

Comment (shadmansaleh) [reply lua/lualine/themes/solarized.lua:31]: acknoledged

File: lua/lualine/themes/solarized.lua (hunk 1 above)
Comment (shadmansaleh) [reply lua/lualine/themes/solarized.lua:27]: acknoledged

File: lua/lualine/themes/solarized.lua (hunk 2 above)
Comment (shadmansaleh) [reply lua/lualine/themes/solarized.lua:31]: Well I added it to see if it works . At lest didn't work on my device . I was gona remove it.

**Comment by shadmansaleh** [comment]
Can you tell me if solarized dark and light look different ? they look same on my device . even lightlines seem same . maybe something is overriding my configs.

**Comment by hoob3rt** [comment]
> Can you tell me if solarized dark and light look different ? they look same on my device . even lightlines seem same . maybe something is overriding my configs.

they are similar but different. I don't know if they should be like that because I despise solarized :laughing: Will defo check all your pr's in depth later <3

**Comment by shadmansaleh** [comment]
>they are similar but different. I don't know if they should be like that because I despise solarized

I donn't use solarized either so I donn't know how it should look either.

File: lua/lualine/themes/solarized_light.lua [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  blue    =  '#268bd2',
+  cyan    =  '#2aa198',
+  green   =  '#859900',
+}
+if vim.o.background == 'light' then
+  colors.base03, colors.base3  =  colors.base3, colors.base03
+  colors.base02, colors.base2  =  colors.base2, colors.base02
+  colors.base01, colors.base1  =  colors.base1, colors.base01
+  colors.base00, colors.base0  =  colors.base0, colors.base00
+end
```
Comment (andreacastellanos) [inline lua/lualine/themes/solarized_light.lua:33]: in regards to this [comment](https://github.com/hoob3rt/lualine.nvim/pull/63#issuecomment-759161077)
> Can you tell me if solarized dark and light look different ? they look same on my device . even lightlines seem same . maybe something is overriding my configs.

if you remove this `if` block, the colors will be correct for the light theme. you already inverted the color definitions in `local colors = {...` 
eg:
- `light: base3  =  '#002b36', base03   =  '#fdf6e3',`
- ` dark:  base03  =  '#002b36', base3   =  '#fdf6e3',`

i've been using this branch w that modification and both themes are working correctly

Comment (shadmansaleh) [reply lua/lualine/themes/solarized_light.lua]: I had absulutely forgotten about that . Thanks for letting me know .

Comment (andreacastellanos) [reply lua/lualine/themes/solarized_light.lua]: 👍  thanks for porting this over, it looks and works great!

**Pull request merged**